ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixFactory.java

```
@@ -96,7 +101,14 @@ public DataNodeConfigSource getDataNodeConfigSource(ClusterMapConfig clusterMapC
    * @return a new instance of {@link HelixManager}.
    */
   HelixManager buildZKHelixManager(String clusterName, String instanceName, InstanceType instanceType, String zkAddr) {
```

instanceName will also have port appended https://github.com/linkedin/ambry/blob/6683f022ee4b740d1d5eb90fa5207c2261a7c8a5/ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixFactory.java#L116
May be you can use that and throw error when port is not present.